helixlauncher-core/src/config.rs

```diff
@@ -29,6 +30,9 @@ pub struct Config {
     libraries_dir: PathBuf,
     #[serde(default = "assets_default")]
     assets_dir: PathBuf,
+
+    #[serde(default = "meta_url_default")]
+    meta_url: String,
```

Comment:
I don't. The default should always be listed in the config

Reply:
In the future, it might be possible (and not *too* difficult) to make it so that multiple component sources are included at once. (Similarly to how `get_component_meta` already falls back to a local version) But thats a discussion for another place. For the time being I agree with Ecorous

Reply:
This especially is the kind of default that _can_ change, and isn't too unlikely to change. In fact, I think it may make sense to do so soon, as currently we have no version in the URL.
If the user didn't touch this sort of setting, then why would the old default stay? That seems like fairly unexpected behavior to me.
Falling back to a local version as it's currently implemented is a crude form of caching with offline mode support, and at least by me not intended to be used to add custom component jsons. That should be instance level just like with MultiMC, otherwise you'd have two parts to copy around.

Reply:
> This especially is the kind of default that _can_ change, and isn't too unlikely to change. In fact, I think it may make sense to do so soon, as currently we have no version in the URL. If the user didn't touch this sort of setting, then why would the old default stay? That seems like fairly unexpected behavior to me. Falling back to a local version as it's currently implemented is a crude form of caching with offline mode support, and at least by me not intended to be used to add custom component jsons. That should be instance level just like with MultiMC, otherwise you'd have two parts to copy around.

I think we can just write some config migration code. e.g. `if meta_url == old_urlL; meta_url = new_url`